### PR TITLE
Remove lowering functions and forklift

### PR DESCRIPTION
**Pull request by isovector**

### CODE REVIEW

File: src/Polysemy/Async.hs [hunk 1]
```diff
@@ -48,43 +46,6 @@ sequenceConcurrently :: forall t r a. (Traversable t, Member Async r) =>
...
-          pureT $ inspect ins <$> fa
-
-        Await a -> pureT =<< embed (A.wait a)
-        Cancel a -> pureT =<< embed (A.cancel a)
-    )  m
-{-# INLINE asyncToIO #-}
-
 ------------------------------------------------------------------------------
 -- | Run an 'Async' effect in terms of 'A.async' through final 'IO'.
 --
```
Comment (KingoftheHomeless) [inline src/Polysemy/Async.hs:51]: Documentation for `asyncToIOFinal` must be updated.

File: src/Polysemy/Resource.hs [hunk 2]
```diff
@@ -138,42 +136,6 @@ resourceToIOFinal = interpretFinal $ \case
 {-# INLINE resourceToIOFinal #-}
```
Comment (KingoftheHomeless) [inline src/Polysemy/Resource.hs:136]: The documentation for `resourceToIOFinal` must be updated.

File: src/Polysemy.hs [hunk 3]
```diff
@@ -120,9 +120,6 @@ module Polysemy
   , reinterpret2H
   , reinterpret3H
 
-    -- * Combinators for Interpreting Directly to IO
-  , withLowerToIO
-
     -- * Kind Synonyms
   , Effect
   , EffectRow
```
Comment (KingoftheHomeless) [inline src/Polysemy.hs:125]: `.@`, `.@@` should be removed.

**Review by KingoftheHomeless — CHANGES_REQUESTED** [review]
The documentation needs a pass -- these are the ones I've been able to spot, but I suspect we've been mentioning lowering/forklift related stuff all over the place, and we should ensure such references go down the memory hole.

**Comment by isovector** [comment]
@KingoftheHomeless PTAL. You were right, looking for the documentation to update found a few more things I should remove.

**Comment by isovector** [comment]
I plan to merge this tomorrow unless there are any objections

**Pull request merged**